File: GO-CODE/BubbleSort.go
package main

import "fmt"


func main(){
	array := []int{4,5,3,6,2,7,9,8,1}

for j := 0; j<len(array) - 1;j++{
for i := 0; i<len(array) - 1;i++{
		if(array[i] > array[i + 1]){
			var temp = array[i]
			array[i] = array[i + 1]
			array[i + 1] = temp
		}
	}
}
	
	fmt.Println(array)
}

File: sorting/BubbleSort.java
package JAVA.sorting;

public class BubbleSort {
    static int[] array = new int[]{4,3,2,1,5};
    public static void main(String[] args) {
        sort(array,array.length);
        for(int data: array){
            System.out.println(data);
        }
    }

    static void sort(int[]  array, int len){
        for(int i=0;i<len;i++){

            for(int j=i+1;j<len;j++){
                int eleCurr = array[i];
                int eleNext = array[j];
                if(eleCurr > eleNext){
                    swap(array,i,j);
                }

            }
        }
    }


    static void swap(int[] array,int i,int j){
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }
}
